client/Components/map.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useRef, useCallback, useContext, useEffect } from 'react';
+import EggIcon from '../UI/egg-icon';
+import MapGL, { Marker, GeolocateControl } from 'react-map-gl';
+import Geocoder from 'react-map-gl-geocoder';
+import Navbar from './navbar';
+import CreateEgg from './createEggModal';
+import { UserContext } from '../Context/userContext';
+
+const MAPBOX_TOKEN = 'pk.eyJ1IjoibXBvd2VyczExMTMiLCJhIjoiY2t4Mmo3MTl1MWlyNzJucDJ1d2xtZXA4aiJ9.w4OiI4e8mDRubQC-0vJs3g';
+
+const Map2 = props => {
+
+  const user = useContext(UserContext);
+  const [eggMarkers, setEggMarkers] = useState([]);
+  const [eggLocation, setEggLocation] = useState(null);
+
+  const clearEggData = () => setEggLocation(null);
+
+  const dropEgg = () => {
+    setEggMarkers([...eggMarkers, eggLocation]);
+    clearEggData();
+  };
+
+  useEffect(() => {
+    const getEggs = () => {
+      if (!user) return;
+      fetch('/api/egg', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ userId: user.id })
+      }).then(res => {
+        if (!res.ok) throw new Error('something went wrong fetching user data');
+        return res.json();
+      }).then(res => {
+        const eggs = [];
+        for (const egg of res) {
+          const eggLocation = { id: +egg.Id, longitude: +egg.longitude, latitude: +egg.latitude };
+          eggs.push(eggLocation);
+        }
+        if (eggs.length === 0) return;
```

Comment:
You can leave this conditional check out or put it at the top of your code block